tests/unit/test_agw_network_configurator.py

```diff
@@ -0,0 +1,1037 @@
+#!/usr/bin/env python3
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import os
+import time
+import unittest
+from unittest.mock import Mock, PropertyMock, call, mock_open, patch
+
+from magma_access_gateway_installer.agw_network_configurator import (
+    AGWInstallerNetworkConfigurator,
+)
+
+
+class TestAGWInstallerNetworkConfigurator(unittest.TestCase):
+    TEST_NETWORK_INTERFACES = ["eth0", "eth1"]
+
+    def setUp(self) -> None:
+        self.timestamp = time.time()
+
+    @patch(
+        "magma_access_gateway_installer.agw_network_configurator.AGWInstallerNetworkConfigurator.INTERFACES_DIR"  # noqa: E501, W505
+    )
+    @patch("builtins.open", new_callable=mock_open())
+    def test_given_agw_installer_network_configurator_when_prepare_interfaces_configuration_dir_if_doesnt_exist_then_configuration_dir_gets_created_properly(  # noqa: E501
+        self, mocked_open_file, ifaces_dir
+    ):
+        ifaces_dir.return_value = f".tox/.tmp/interfaces.d_{self.timestamp}"
+        AGWInstallerNetworkConfigurator(
+            self.TEST_NETWORK_INTERFACES, None, None
+        )._prepare_interfaces_configuration_dir_if_doesnt_exist()
+        self.assertTrue(os.path.exists(ifaces_dir))
```

Comment:
Are we really creating a file in`.tox/.tmp/interfaces`?

Reply:
Test changed to just check whether `os.makedirs` has been called with correct params.